src/srv.rs

```diff
@@ -0,0 +1,140 @@
+
+use types::{
+    AresError,
+};
+use utils::ares_error;
+
+/// The result of a successful lookup for an SRV record.
+pub struct SRVResults {
+    srv_reply: *mut c_ares_sys::Struct_ares_srv_reply,
+}
+
+/// The contents of a single SRV record.
+pub struct SRVResult<'a> {
+    // A single result - reference into an `SRVResults`.
+    srv_reply: *mut c_ares_sys::Struct_ares_srv_reply,
+    phantom: PhantomData<&'a SRVResults>,
+}
+
+impl SRVResults {
+    /// Obtain an `SRVResult` from the response to an SRV lookup.
+    pub fn parse_from(data: &[u8]) -> Result<SRVResults, AresError> {
+        let mut srv_reply: *mut c_ares_sys::Struct_ares_srv_reply = ptr::null_mut();
+        let parse_status = unsafe {
+            c_ares_sys::ares_parse_srv_reply(
+                data.as_ptr(),
+                data.len() as libc::c_int,
+                &mut srv_reply as *mut *mut _ as *mut *mut c_ares_sys::Struct_ares_srv_reply)
+        };
+        if parse_status != c_ares_sys::ARES_SUCCESS {
+            Err(ares_error(parse_status))
+        } else {
+            let srv_result = SRVResults::new(srv_reply);
+            Ok(srv_result)
+        }
+    }
+
+    fn new(srv_reply: *mut c_ares_sys::Struct_ares_srv_reply) -> SRVResults {
+        SRVResults {
+            srv_reply: srv_reply,
+        }
+    }
+
+    /// Returns an iterator over the `SRVResult` values in this `SRVResults`.
+    pub fn iter(&self) -> SRVResultsIterator {
+        SRVResultsIterator {
+            next: unsafe { (*self.srv_reply).next },
+            phantom: PhantomData,
+        }
+    }
+}
+
+pub struct SRVResultsIterator<'a> {
+    next: *mut c_ares_sys::Struct_ares_srv_reply,
+    phantom: PhantomData<&'a SRVResults>,
+}
+
+impl<'a> Iterator for SRVResultsIterator<'a> {
+    type Item = SRVResult<'a>;
+    fn next(&mut self) -> Option<Self::Item> {
+        let srv_reply = self.next;
+        if srv_reply.is_null() {
+            None
+        } else {
+            self.next = unsafe { (*srv_reply).next };
+            let srv_result = SRVResult {
+                srv_reply: srv_reply,
+                phantom: PhantomData,
+            };
+            Some(srv_result)
+        }
+    }
+}
+
+impl<'a> IntoIterator for &'a SRVResults {
+    type Item = SRVResult<'a>;
+    type IntoIter = SRVResultsIterator<'a>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        SRVResultsIterator {
+            next: unsafe { (*self.srv_reply).next },
+            phantom: PhantomData,
+        }
+    }
+}
+
+impl Drop for SRVResults {
+    fn drop(&mut self) {
+        unsafe {
+            c_ares_sys::ares_free_data(self.srv_reply as *mut libc::c_void);
+        }
+    }
+}
+
+impl<'a> SRVResult<'a> {
+    /// Returns the hostname in this `SRVResult`.
+    pub fn host(&self) -> &str {
+        unsafe {
+            let c_str = CStr::from_ptr((*self.srv_reply).host);
+            str::from_utf8(c_str.to_bytes()).unwrap()
+        }
+    }
+
+    pub fn weight(&self) -> u16 {
```

Comment:
Please add comments for all public functions.  These get autogenerated into documentation - see [here](http://dimbleby.github.io/rust-c-ares)